### ISSUE TITLE
net.Socket js Cannot cast T to <function> String 

### ISSUE THREAD

**Issue by MetaChrome** [openfl/openfl#1119]
#if (js && html5)
    __socket.send (cast (__output, ArrayBuffer));

At approximately line 186, is yielding an exception of the following nature:

    Cannot cast T to <function> String 

Is there a way to fix this? This is with haxe 3.2.1

**Comment by MetaChrome** [comment]
@:to ByteArray.toBytesData at ByteArray line 153 should probably be called but it's not. It's private but setting it to public doesn't work. I'm not sure if cast(a,B) calls the @:to or if that only occurs in a assignment to a target Typed instance. Either way, doing 

    var output_data:ByteArrayData=__output;
    __socket.send (output_data.getData());

fixed this.

**Comment by MSGhero** [comment]
What version of openfl? There were some restructures around byte arrays not too long ago.

**Comment by MetaChrome** [comment]
well this is on master again

**Comment by MetaChrome** [comment]
openfl is master, haxe is latest provided release. ill test against haxe master shortly i suppose. i'd have thought it would be apparent if this is a mistake or not

**Comment by MSGhero** [comment]
I just tried in try.haxe.org, and it doesn't seem like casting works. But yeah, that line forgot the `getData()` call or it could just make use of the abstract instead of fighting it with a cast.

**Comment by troyc** [comment]
Using this code
```var output_data:ByteArrayData=__output;

__socket.send (output_data.getData());```
works, but it seems to send extra junk data (null bytes) at the end of each message.

Using this code
```__socket.send (__output);```
also works, but it always tries to convert the data to a string so the message gets cut off at the first null byte.

**Comment by troyc** [comment]
This was just resolved by commits 16735289efbdd22c89d4e1da2fe2654cdf97b140 15a39fd043568ff2450bfbb7ff37cadfd451e67f 286fb2fa02fd7b88c4e94b5fbde8015179ed51e2
